### DISCUSSION TITLE
How to set CWD as the root_dir for markdown oxide language server?

### DISCUSSION THREAD

**Discussion by Feel-ix-343** [helix-editor/helix#10562] [Q&A]
Hello Helix community, from [this issue](https://github.com/Feel-ix-343/markdown-oxide/issues/44), I realize that when a git repo is not initialized, helix passes nothing to the LSP root_dir config field for the markdown oxide language server. Is it possible to change the config for this language server to specify that CWD should be passed in?

**Comment by pascalkuthe** [comment]
this is intentional. Most language servers take the presence of a root dir to mean that it was started inside somekind of project directory that it should index.

That is undesirable if there are no root marker or git repo because that would mean if helix is started in `$HOME` or `/` then the lsp would scan almost the entire disk.

You can add `.obsidian` or `.moxide.toml` as root markers for that lsp. But in the case the lsp should correctly interpret a null value. Its an intentional feature of LSP that this is an optional field.

If CWD makes the most sense for `markdown_oxide` then they should use `CWD` if initialized with the roo set to `null`

**Reply by Feel-ix-343** [reply]
OK

I implemented this for the language server. 

However in neovim, I am finding that the editor's file watcher is disabled when root_dir is not passed (putting nvim-lspconfig into single-file-mode) -- making it preferable to specify root_dir as CWD in nvim-lspconfig

Are there any limitations like this in helix?

**Reply by pascalkuthe** [reply]
We don't have a filewatcher yet so right now no but I would imagine we may do the same in the future. I was actually planning to disable it outside of git repos entirely since scanning all of home or the fs root is not a good idea.

I think what you are doing in nvim lap config will cause a bunch of lag if you open a markdown file in a large directory like $HOME. 

Passing null is essentially single file mode since that only happens when outside a git repository and no root marker was found.

To be clear I don't know what nvim does internally but we would always watch open buffers but we wouldn't be watching all files in the directory and scan for changes using the lsp provided glob.

If you don't have any root markers then that shouldn't be a problem I think since lsp mostly uses the file watcher to check for configuration changes (and the same config could be used as a root marker)
